aamt_deploysite.pl

```diff
@@ -0,0 +1,159 @@
+    }
+    
+
+    &pars_FirstPass($localConfFilePath);    # Parser Module first pass
+    &pars_SetRecoveryCode(RECOVERY_MODE_1);
+    
+    # Concatenate all of the configuration files in preparation for the next step where we parse the master file
+    use Cwd;
+    my $pwd = cwd();
+    # get the current working folder
+    my $strCurWorkingFolder = &utf_getCurrentWorkingFolder();
+    # get session name
+    my $strSessionName = &ilog_getSessionName();
+    # form the complete working folder
+    my $workingFolder = $strCurWorkingFolder . '/' . $strSessionName;
+    # change local dir
+    my $retwrk_changeLocalDir = wrk_changeLocalDir($workingFolder);
+    if (!($retwrk_changeLocalDir))
+    {
+        $logFileReturn= ilog_setLogInformation('EXT_ERROR',ERR_CWD_COMMAND,'', __LINE__);
+        if(!($logFileReturn))
+       {	
+           $logFileReturn=ilog_print(ERR_INTERNAL_ERROR_CONSOLE.__LINE__,1);
+        }
+
+        return 0;
+    }
+
+    # my @files        = grep { -f } glob( '*.conf' );
+    my @files = File::Find::Rule->file()
+        ->name("*apache*")            
+        ->in($workingFolder);
+
+	if(! @files)
+	{
+	    @files = File::Find::Rule->file()
+       	    ->name("*httpd*")            
+            ->in($workingFolder);
+	}
+
+    my @fhs          = map { open my $fh, '<', $_; $fh } @files;
+    my $concatenated = '';
+    while (my $fh = shift @fhs) 
+    {
+        while ( my $line = <$fh> )
+        {
+            $concatenated .= $line;
+        }
+
+        close $fh;
+    }
+
+    # go back to orginal dir
+    chdir($pwd); 
+
+    my $confAllName = &utf_getCompleteFilePath(FILE_CONF_ALL);
+    my $HANDLE_CONF_ALL = new IO::File;
+    if(open(HANDLE_CONF_ALL,">> $confAllName") or die 'ERR_FILE_OPEN')
+    {
+        print HANDLE_CONF_ALL $concatenated;
+        close(HANDLE_CONF_ALL); 
+    }
+     
+    my $logFilereturn = ilog_setLogInformation('REC_INFO',"FILE_CONF_ALL ".REC_ADD_EQUAL,&utf_getCompleteFilePath(FILE_CONF_ALL),'');
+    &pars_SetRecoveryCode(RECOVERY_MODE_2);
+
+    if ($DEBUG_MODE) { ilog_print(1,"\nDEBUG: RECOVERY MODE: [$RecoveryMode]\n"); }
+    &pars_Generate2D(&utf_getCompleteFilePath(FILE_CONF_ALL), &utf_getCompleteFilePath(FILE_RECOVERY), $RecoveryMode,$ARGV[2]);
+
+	if(! &pars_IsNoSiteSelected()) { &pars_UploadPublishSettingsAllSites($ARGV[3]);}
+
+    &pars_SetRecoveryCode(RECOVERY_MODE_3);
+    &utf_setCurrentModuleName(''); 
+    &utf_gettimeinfo('1');
+    &DeleteWorkingFolder();
+    &TerminateTool();
+};
+if($@)
+{
+    #if ($@ !~ /EXIT_TOOL_NOW/)
+    {        
+        # Abnormal Termination ... 
+        print "$@";
+        &DeleteWorkingFolder();
+    }
+}
+# end of main subroutine
+
+sub TerminateTool
+{
+    # Clean up and exit tool
+    &utf_DisposeFiles();
+    # Restore the include to the original value
+    @INC = @lib::ORIG_INC;
+    exit(0);
+}
+
+sub DeleteWorkingFolder
+{
+    my $strYesOrNo = "";
+#    while($strYesOrNo!~/^\s*[YynN]\s*$/)
```

Comment:
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip retries, if it is false, then keep the current logic of asking the user for input.